test/e2e/operator.go

```diff
@@ -344,41 +344,43 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 		// This is expensive but will prevent flakes.
 		By("gathering existing subnet NSGs")
 		for subnet := range subnetsToReconcile {
-			subnetObject, err := clients.Subnet.Get(ctx, resourceGroup, vnetName, subnet, "")
+			subnetObject, err := clients.Subnet.Get(ctx, resourceGroup, vnetName, subnet, nil)
 			Expect(err).NotTo(HaveOccurred())
```

Comment:
We don't have to assign the response to `resp` to get `Subnet` every time.

The first return value of Get is SubnetsClientGetResponse.
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork#SubnetsClientGetResponse
and it can transparently reference to the fields of Subnet like this.
```
subnetsToReconcile[subnet] = subnetObject.Properties.NetworkSecurityGroup.ID
```
So we don't have to use `resp` just to get the `Subnet` value.
However I used `resp` in some places where I have to reassign the value to `Subnet` type, otherwise I don't use `resp` and use it directly as if it is a `Subnet` type.

Reply:
Thank you for the clarification.